pkg/exchange/okex/exchange.go

```diff
@@ -411,6 +415,126 @@ func (e *Exchange) QueryOrderTrades(ctx context.Context, q types.OrderQuery) ([]
+If you want to query orders by time range, please just pass since and until.
+If you want to query by cursor, please pass lastOrderID.
+Because it gets the correct response even when you pass all parameters with the right time interval and invalid lastOrderID, like 0.
+Time interval boundary unit is second.
+since is inclusive, ex. order created in 1694155903, get response if query since 1694155903, get empty if query since 1694155904
+until is not inclusive, ex. order created in 1694155903, get response if query until 1694155904, get empty if query until 1694155903
+*/
+func (e *Exchange) QueryClosedOrders(ctx context.Context, symbol string, since, until time.Time, lastOrderID uint64) ([]types.Order, error) {
+	if symbol == "" {
+		return nil, ErrSymbolRequired
+	}
+
+	if err := orderRateLimiter.Wait(ctx); err != nil {
+		return nil, fmt.Errorf("query closed order rate limiter wait error: %w", err)
+	}
+
+	var lastOrder string
+	if lastOrderID <= 0 {
+		lastOrder = ""
+	} else {
+		lastOrder = strconv.FormatUint(lastOrderID, 10)
+	}
+
+	res, err := e.client.NewGetOrderHistoryRequest().
+		InstrumentID(toLocalSymbol(symbol)).
+		StartTime(since).
+		EndTime(until).
+		Limit(defaultQueryLimit).
+		Before(lastOrder).
+		Do(ctx)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to call get order histories error: %w", err)
+	}
+
+	var orders []types.Order
+
+	for _, order := range res {
+		o, err2 := toGlobalOrder(&order)
+		if err2 != nil {
+			err = multierr.Append(err, err2)
+			continue
+		}
+
+		orders = append(orders, *o)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return types.SortOrdersAscending(orders), nil
+}
+
+/*
+QueryTrades can query trades in last 3 months, there are no time interval limitations, as long as end_time >= start_time.
+OKEX do not provide api to query by tradeID, So use /api/v5/trade/orders-history-archive as its official site do.
+If you want to query trades by time range, please just pass start_time and end_time.
+Because it gets the correct response even when you pass all parameters with the right time interval and invalid LastTradeID, like 0.
+*/
+func (e *Exchange) QueryTrades(ctx context.Context, symbol string, options *types.TradeQueryOptions) ([]types.Trade, error) {
+	if symbol == "" {
+		return nil, ErrSymbolRequired
+	}
+
+	if options.LastTradeID > 0 {
+		log.Warn("!!!OKEX EXCHANGE API NOTICE!!! Okex does not support searching for trades using TradeId.")
+	}
+
+	req := e.client.NewGetOrderHistoryRequest().InstrumentID(toLocalSymbol(symbol))
+
+	limit := uint64(options.Limit)
+	if limit > defaultQueryLimit || limit <= 0 {
+		log.Debugf("limit is exceeded default limit %d or zero, got: %d, Do not pass limit", defaultQueryLimit, options.Limit)
+	} else {
+		req.Limit(limit)
+	}
+
+	if err := orderRateLimiter.Wait(ctx); err != nil {
+		return nil, fmt.Errorf("query trades rate limiter wait error: %w", err)
+	}
 
+	var err error
+	var response []okexapi.OrderDetails
```

Comment:
I don't understand, you're calling QueryOrderHistory in the QueryTrades api? is this expected?

Reply:
Yes ! because OKX didn't provide any api query by trade_id, but QueryOrderHistory has trade_id in its response. So I call QueryOrderHistory to get the same effect.

Reply:
OKX didn't provide any api query like `/my-trade` either

Reply:
你可以測試pagination是否可以動嗎?

Reply:
you are right, 加了unit test 後發現應該要用 `limit` 而不是 `defaultQueryLimit`